### PR TITLE
Update fastlane side of ActionCollector to support version and crash

### PR DESCRIPTION
**Pull request by mfurtak**
- Bump minimum required version of `fastlane_core` for `ToolCollector` changes
- Remove String monkey-patching from `fastlane` that has been moved into `fastlane_core`
- Update `cli_tools_distributor` to use the new `fastlane_module` String method in `fastlane_core`
- Fix `cli_tools_distributor_spec` to be runnable on its own
- Fix `puts` action to not error if passed a `nil` value
- Update `fastlane_runner` to distinguish between a user error and a crash

### CODE REVIEW

File: fastlane/spec/cli_tools_distributor_spec.rb [hunk 1]
```diff
@@ -2,15 +2,16 @@
 
 describe Fastlane::CLIToolsDistributor do
   it "runs the lane instead of the tool when there is a conflict" do
-    require 'fastlane/commands_generator'
     ARGV = ["sigh"]
+    require 'fastlane/commands_generator'
```
Comment (KrauseFx) [inline fastlane/spec/cli_tools_distributor_spec.rb]: Any reason why this was changed?

**Comment by KrauseFx** [comment]
Looks great, really good changes!

File: fastlane/spec/cli_tools_distributor_spec.rb (hunk 1 above)
Comment (mfurtak) [inline fastlane/spec/cli_tools_distributor_spec.rb]: That was for: "Fix cli_tools_distributor_spec to be runnable on its own"

If you go to master and run `bundle exec rspec spec/cli_tools_distributor_spec.rb` you should find that it has a failure.

Comment (mfurtak) [inline fastlane/spec/cli_tools_distributor_spec.rb]: I just made the ordering consistent with the other test :smile:

**Pull request merged**